* Solaris doesn't have flaky and pretend python packages available.

* Since blowfish is part of the legacy provider in OpenSSL 3, it's changed
  to DES-EDE3-CBC available in the default one. This wasn't reported upstream
  because standard cryptography doesn't work without the legacy provider, and
  thus this shouldn't be an issue (but our patched variant doesn't need it).

* test_set_default_verify_paths tries to connect to an external server, which
  does not work within OWAN, even if proxy variables are set.

--- pyopenssl-24.2.1/tests/test_crypto.py
+++ pyopenssl-24.2.1/tests/test_crypto.py
@@ -78,7 +78,7 @@ def utcnow():
     return datetime.now(timezone.utc).replace(tzinfo=None)
 
 
-GOOD_CIPHER = "blowfish"
+GOOD_CIPHER = "DES-EDE3-CBC"
 BAD_CIPHER = "zippers"
 
 GOOD_DIGEST = "SHA256"
@@ -1995,7 +1995,6 @@ class TestX509(_PKeyInteractionTestsMixi
         with pytest.raises(TypeError):
             cert.gmtime_adj_notBefore(None)
 
-    @pytest.mark.flaky(reruns=2)
     def test_gmtime_adj_notBefore(self):
         """
         `X509.gmtime_adj_notBefore` changes the not-before timestamp to be the
@@ -2021,7 +2020,6 @@ class TestX509(_PKeyInteractionTestsMixi
         with pytest.raises(TypeError):
             cert.gmtime_adj_notAfter(None)
 
-    @pytest.mark.flaky(reruns=2)
     def test_gmtime_adj_notAfter(self):
         """
         `X509.gmtime_adj_notAfter` changes the not-after timestamp
--- pyopenssl-24.2.1/tests/test_ssl.py
+++ pyopenssl-24.2.1/tests/test_ssl.py
@@ -39,7 +39,6 @@ from cryptography import x509
 from cryptography.hazmat.primitives import hashes, serialization
 from cryptography.hazmat.primitives.asymmetric import rsa
 from cryptography.x509.oid import NameOID
-from pretend import raiser
 
 from OpenSSL import SSL
 from OpenSSL._util import ffi as _ffi
@@ -509,7 +508,6 @@ class TestContext:
         with pytest.raises(TypeError):
             context.set_cipher_list(object())
 
-    @pytest.mark.flaky(reruns=2)
     def test_set_cipher_list_no_cipher_match(self, context):
         """
         `Context.set_cipher_list` raises `OpenSSL.SSL.Error` with a
@@ -845,7 +843,7 @@ class TestContext:
         """
         key = PKey()
         key.generate_key(TYPE_RSA, 1024)
-        pem = dump_privatekey(FILETYPE_PEM, key, "blowfish", passphrase)
+        pem = dump_privatekey(FILETYPE_PEM, key, "DES-EDE3-CBC", passphrase)
         with open(tmpfile, "w") as fObj:
             fObj.write(pem.decode("ascii"))
         return tmpfile
@@ -1221,6 +1219,7 @@ class TestContext:
         assert context._check_env_vars_set(dir_var, file_var) is True
         assert context._check_env_vars_set(dir_var, file_var) is True
 
+    @pytest.mark.skip(reason="We don't have pretend python package in Solaris")
     def test_verify_no_fallback_if_env_vars_set(self, monkeypatch):
         """
         Test that we don't use the fallback path if env vars are set.
@@ -1249,6 +1248,9 @@ class TestContext:
         reason="set_default_verify_paths appears not to work on Windows.  "
         "See LP#404343 and LP#404344.",
     )
+    @pytest.mark.skip(
+        reason="This test doesn't work within OWAN, even with proxy set"
+    )
     def test_set_default_verify_paths(self):
         """
         `Context.set_default_verify_paths` causes the platform-specific CA
